activemq-broker/src/main/java/org/apache/activemq/broker/region/policy/AbstractDeadLetterStrategy.java

```diff
@@ -108,20 +111,25 @@ public void setExpiration(long expiration) {
         this.expiration = expiration;
     }
 
+
+
     public int getMaxProducersToAudit() {
-        return messageAudit.getMaximumNumberOfProducersToTrack();
+        return this.maxProducersToAudit;
     }
 
     public void setMaxProducersToAudit(int maxProducersToAudit) {
-        messageAudit.setMaximumNumberOfProducersToTrack(maxProducersToAudit);
+        this.maxProducersToAudit = maxProducersToAudit;
     }
 
     public void setMaxAuditDepth(int maxAuditDepth) {
-        messageAudit.setAuditDepth(maxAuditDepth);
+        this.maxAuditDepth = maxAuditDepth;
```

Comment:
This sets the variable but no longer sets the shared audit so that will break things.